### PR TITLE
fix: improve time rules, fix failing tests

### PR DESCRIPTION
**Pull request by scroskey-clinc**
Fix time rule issues:
- Add rule for "the \<day-of-month\> of \<time\>
- Add "starting" and "ending" to date range rules
- Require non-latent number for intersections between a number and time
- Update corpora to fix test suites

### CODE REVIEW

File: Duckling/Time/EN/Corpus.hs [hunk 1]
```diff
@@ -163,6 +163,14 @@ latentCorpus = (testContext, testOptions {withLatent = True}, xs)
...
              ,  "twelve oh three"
              ]
+      , examples (datetime (2013, 2, 12, 4, 30, 0) Second)
+             [ "now"
+             , "right now"
+             , "just now"
+             ]
+      , examples (datetime (2013, 2, 15, 0, 0, 0) Day)
+             [ "on the 15th"
+             ]
```
Comment (wang8zai) [inline Duckling/Time/EN/Corpus.hs:173]: The reason to make these corpus latent is that these datetime may not only belong to time dim? Not sure if I understand this correct?

File: tests/Duckling/Time/Tests.hs [hunk 2]
```diff
@@ -82,8 +82,8 @@ timeFormatTest :: TestTree
 timeFormatTest = testCase "Format Test" $
   mapM_ (analyzedFirstTest testContext testOptions . withTargets [Seal Time]) xs
   where
-    xs = examplesCustom (parserCheck expected parseValue) ["now"]
-    expected = "2013-02-12T04:30:00.000-02:00"
```
Comment (wang8zai) [inline tests/Duckling/Time/Tests.hs:86]: So I think we on purposely do not want to handle "now" using duckling right?

File: tests/Duckling/AmountOfMoney/EN/Tests.hs [hunk 3]
```diff
@@ -144,7 +144,7 @@ intersectTests = testCase "Intersect Test" $
          , ("1.1 dollars 6", 2)
          , ("10 cents and 8 cents", 2)
          , ("10 dollars and 2 dollars", 2)
-         , ("between 4.1 dollars and 4 dollars", 3)
+         , ("between 4.1 dollars and 4 dollars", 2)
```
Comment (wang8zai) [inline tests/Duckling/AmountOfMoney/EN/Tests.hs:147]: Does this means the body we classified from "between 4.1 dollars and 4 dollars" was "4.1 dollars", "dollars and 4", and "4 dollars"; and it becomes "4.1 dollars" and "4 dollars" after we change the rules? Just not sure how the 3 and 2 comes from.

File: Duckling/Time/EN/Corpus.hs (hunk 1 above)
Comment (scroskey-clinc) [reply Duckling/Time/EN/Corpus.hs:173]: Yeah, that's correct. This is one of the time patterns which are too ambiguous to treat as non-latent without more context

File: tests/Duckling/Time/Tests.hs (hunk 2 above)
Comment (scroskey-clinc) [reply tests/Duckling/Time/Tests.hs:86]: Yeah I made "now" latent with a previous release. The problem is that it is used often on updates from the user outside of the context of a time related to the query: 

1. "show my transactions from Walmart last month"
2. "now show my transactions from Kroger"

We'd want to maintain "last month" in this case. "now" also tends to be a default for actions (e.g. transfers) so it's safer to not extract than to extract erroneously

**Review by wang8zai — APPROVED** [review]

File: tests/Duckling/AmountOfMoney/EN/Tests.hs (hunk 3 above)
Comment (scroskey-clinc) [reply tests/Duckling/AmountOfMoney/EN/Tests.hs:147]: I think that's accurate. I believe that the test case is tracking the number of unique parses for the input where token ranges which are subsets of another token range are excluded:

- "4.1 dollars"
- "4 dollars"
- "1 dollars and 4"

My change removed the rule which allowed for the "1 dollars and 4" parse.

The classifier seems to only be taken into consideration for figuring out the prioritization for applying rules for identical token ranges:

- ((after march) 2019)
- (after (march 2019))

**Pull request merged**